crates/gpui/src/platform/windows/window.rs

```diff
@@ -158,6 +159,12 @@ impl WindowsWindowInner {
         }
     }
 
+    /// mark window client rect to be re-drawn
+    /// https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-invalidaterect
+    pub(crate) fn invalidate_client_area(&self) {
+        unsafe { InvalidateRect(self.hwnd, None, FALSE) };
```

Comment:
are we sure this is going to make the window refresh with the display?
reading this, it sounds more like refreshing at max speed:
>The system sends a [WM_PAINT](https://learn.microsoft.com/en-us/windows/desktop/gdi/wm-paint) message to a window whenever its update region is not empty and there are no other messages in the application queue for that window.

Reply:
ah you're right. I just tested again and it's rendering as fast as possible. I read online where someone else was claiming that and I either misunderstood or it was flat out wrong.

Reply:
should be better now. I refactored to use `DCompositionWaitForCompositorClock` and brought back the peek message loop.